packages/UI/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,80 @@
+import * as Dialog from '@radix-ui/react-dialog';
+import { X } from 'lucide-react';
+import { tv } from 'tailwind-variants';
+export { Close, Description } from '@radix-ui/react-dialog';
+
+export interface ModalProps {
+  /**
+   * hasCloseBtn: boolean - defaulted true, will render a close button in the top right corner of the modal
+   */
+  hasCloseBtn?: boolean;
+  /**
+   * trigger: React.ReactNode - the element that will trigger the modal to open
+   */
+  trigger: React.ReactNode;
+  /**
+   * header: React.ReactNode - the element that will be rendered in the top of the modal
+   */
+  header: React.ReactNode;
+  /**
+   * content: React.ReactNode - the element that will be rendered in the middle of the modal
+   */
+  content: React.ReactNode;
+  /**
+   * footer: React.ReactNode - the element that will be rendered at the bottom of the modal
+   */
+  footer: React.ReactNode;
+  /**
+   * className: string - the className that will be applied to the root element
+   */
+  className?: string;
+}
+
+/**
+ * @description A modal is a dialog box/popup window that is displayed on top of the current page
+ * https://www.radix-ui.com/primitives/docs/components/dialog
+ * Close export is an wrapper to close the modal from a trigger element.
+ * Description export is an wrapper to add a description to the modal for accessibility.
+ */
+const Modal = ({
+  header,
+  trigger,
+  footer,
+  content,
+  className,
+  hasCloseBtn = true,
+}: ModalProps) => {
+  const { rootStyle, overlayStyle } = modalStyles();
+  return (
+    <Dialog.Root>
+      <Dialog.Trigger asChild>{trigger}</Dialog.Trigger>
+      <Dialog.Portal>
+        <Dialog.Overlay className={overlayStyle()} />
+        <Dialog.Content className={rootStyle({ className })}>
+          <Dialog.Title className="w-full">{header}</Dialog.Title>
+          {hasCloseBtn && (
+            <Dialog.Close asChild>
+              <button
+                className="absolute right-8 top-8 flex h-7 w-7 items-center justify-center rounded-md bg-white drop-shadow-xl"
+                aria-label="Close"
+              >
+                <X />
+              </button>
+            </Dialog.Close>
+          )}
+          <div className="my-6 min-h-36 w-full">{content}</div>
```

Comment:
eslint error: `Classname 'min-h-36' is not a Tailwind CSS class!`

Reply:
@Enjoy2Live it exists in the docs. 
https://tailwindcss.com/docs/min-height

Reply:
@l1keafox indeed, I need to debug the eslint rule

Reply:
does eslint say the same thing on your

Reply:
fixed by updating the eslint plugin as well as tailwind itself. nevermind!